### PR TITLE
INTMDB-604: alert configuration profile changes

### PR DESCRIPTION
**Pull request by govardhanpagidi**
## Description

profile changes for alert configuration 

Link to any related issue(s): 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: cfn-resources/alert-configuration/cmd/resource/model.go [hunk 1]
```diff
@@ -35,19 +21,19 @@ type Model struct {
...
 	Href    *string           `json:",omitempty"`
 	Rel     *string           `json:",omitempty"`
 }
 
+// ApiKeyDefinition is autogenerated from the json schema
+type ApiKeyDefinition struct {
+	PrivateKey *string `json:",omitempty"`
+	PublicKey  *string `json:",omitempty"`
+}
+
```
Comment (andreaangiolillo) [inline cfn-resources/alert-configuration/cmd/resource/model.go:36]: this is not needed

File: cfn-resources/alert-configuration/resource-role.yaml [hunk 2]
```diff
@@ -15,22 +15,36 @@ Resources:
...
+                - "cloudformation:ListResources"
+                - "cloudformation:UpdateResource"
+                - "iam:AttachRolePolicy"
+                - "iam:CreateRole"
+                - "iam:DeleteRole"
+                - "iam:GetRole"
+                - "iam:GetRolePolicy"
+                - "iam:ListAttachedRolePolicies"
+                - "iam:ListRolePolicies"
+                - "iam:PutRolePolicy"
```
Comment (andreaangiolillo) [inline cfn-resources/alert-configuration/resource-role.yaml:47]: why do we need these permissions?

**Review by andreaangiolillo — APPROVED** [review]
LGTM. My comments are not blocking and can be addressed in a follow up PR

File: cfn-resources/alert-configuration/cmd/resource/model.go (hunk 1 above)
Comment (govardhanpagidi) [reply cfn-resources/alert-configuration/cmd/resource/model.go:36]: this APIkey is different. Links Property still uses this APIKeys Definition

**Pull request merged**